### PR TITLE
Enhancement: Support National Clouds

### LINKED ISSUE #16: Support Azure National Clouds
I noticed that AAD endpoint currently is hard coded to Azure public cloud,
and according to https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud

National cloud | Azure AD authentication endpoint
-- | --
Azure AD for US Government | https://login.microsoftonline.us
Azure AD Germany | https://login.microsoftonline.de
Azure AD China operated by 21Vianet | https://login.chinacloudapi.cn
Azure AD (global service) | https://login.microsoftonline.com

I wish we can add an extra config for AAD endpoint (a dropdown or simple input) to support national clouds:)

Actually I made it locally, but I have no idea how to build the code (not familiar with js stuff), or I can try create a PR for you:)

### PR DESCRIPTION
**Pull request by joshhunt**
Closes #16

Adds support for the Azure US Gov (Fairfax) and China (Mooncake) national clouds.

Need to do a bit of light smoke tests first before testing in the other clouds.

### CODE REVIEW

**Review by sunker — APPROVED** [review]
Looks good, but we need to find a way of solving this for already configured data sources.

**Review by sunker — CHANGES_REQUESTED** [review]
Looks good, but we need to find a way of solving this for already configured data sources.

File: src/plugin.json [hunk 1]
```diff
@@ -46,12 +39,12 @@
         }
       ],
       "tokenAuth": {
-        "url": "https://login.microsoftonline.com/{{.JsonData.tenantId}}/oauth2/token",
+        "url": "{{.JsonData.nationalCloud}}/{{.JsonData.tenantId}}/oauth2/token",
```
[the remark this answers is not in the document]
Comment (sunker) [reply src/plugin.json:42]: Yeah. Don't think there's any simple solution to this unfortunately. One path forward though would be to skip the reverse proxy in grafana core and implement CallResourceHandler and handle auth there instead. This is kind of explained [here](https://github.com/grafana/grafana/issues/33809).

[the remark this answers is not in the document]
Comment (joshhunt) [reply src/plugin.json:42]: oh i was hoping there were the hooks to override the URL in the ADX backend code. sad face

[the remark this answers is not in the document]
Comment (floushee) [reply src/plugin.json:42]: Would it be an option to use https://login.microsoftonline.com per default as a fallback? Then there should not be any issues with already configured data sources?

[the remark this answers is not in the document]
Comment (floushee) [reply src/plugin.json:42]: How did they make it work in the https://github.com/grafana/azure-monitor-datasource data source?

**Pull request closed without merging**